lib/sinon/fake-api.js

```diff
@@ -0,0 +1,184 @@
+"use strict";
+
+var arrayProto = require("@sinonjs/commons").prototypes.array;
+var fakeCall = require("./call");
+var callUtil = require("./call-util");
+
+var push = arrayProto.push;
+var forEach = arrayProto.forEach;
+var concat = arrayProto.concat;
+var ErrorConstructor = Error.prototype.constructor;
+var bind = Function.prototype.bind;
+var filter = Array.prototype.filter;
+
+var callId = 0;
+
+// Public API
+var fakeApi = {
```

Comment:
Since this is used by both `spy` and `fake` and there's a name overlap with `fake`, perhaps now is the time to rename this to something different?

`commonApi`, `sharedApi`, `your-suggestion-here`...

Reply:
Good point. I'm not sure. We've been using the term `fake` as the generic term for `spy` and `stub` before, e.g. `instantiateFake`.

Reply:
In this particular case, none of the names including `fake` is part of the public api. Is that correct?

Reply:
I think so, yes.

Reply:
Thinking a bit more about this releaved even more cleanup that I'm about to add.

Reply:
Excellent!